src/eventValidator.js

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+import createError from 'http-errors';
+import Joi from 'joi';
+
+function buildSchema(def) {
+  const body = _.get(def, 'body', {});
+  const headers = _.get(def, 'headers', {});
+  const path = _.get(def, 'path', {});
+  const query = _.get(def, 'query', {});
+
+  if (_.has(def, 'pagination')) {
+    query.offset = Joi.number().integer().default(0)
+      .min(0)
+      .allow('')
+      .error(createError(422, 'Parameter offset must be integer greater then 0'));
+    const max = _.get(def, 'pagination.max', 1000);
+    query.limit = Joi.number().integer().default(max)
+      .min(1)
+      .max(_.get(def, 'pagination.max', max))
+      .allow('')
+      .error(createError(422, `Parameter limit must be integer between 1 and ${max}`));
+  }
+
+  const res = {};
+  if (_.size(body)) {
+    res.body = Joi.object().allow(null).keys(body);
+  }
+  if (_.size(headers)) {
+    res.headers = Joi.object().allow(null).keys(headers);
+  }
+  if (_.size(path)) {
+    res.pathParameters = Joi.object().allow(null).keys(path);
+  }
+  if (_.size(query)) {
+    res.queryStringParameters = Joi.object().allow(null).keys(query);
+  }
+
+  return res;
+}
+
+export default function eventValidator({ schema }) {
+  return {
+    before: (handler, next) => {
+      const res = Joi.validate(
+        handler.event,
+        buildSchema(schema),
+        { allowUnknown: true, stripUnknown: true }
```

Comment:
Co presne robia tie parametre v tejto kombinacii? Stale som to uplne nepochopil, ja pouzivam len `stripUnknown: true` aby sa mi tam nedostavali podivne parametre. Ale ked tam mas aj to `allowUnknown: true` tak ich to myslim nevyhodi..

Reply:
Joi validuje celou tu eventu Lambdy co dostanu na vstupu a díky tomuhle nastavení se ignorují headers a další data který mě pro validaci parametrů nezajímají. Ale je vlastně možný, že je tam to `allowUnknown` zbytečný, když se díky s pomocí `stripUnknown` ty neznámý odstraní.